File: src/main/java/main/json/BooleanJson.java
package main.json;

import org.jetbrains.annotations.NotNull;

/**
 *
 * @author Bunnyspa
 */
public class BooleanJson implements Json {

    private final boolean data;

    public BooleanJson(@NotNull String data) {
        this.data = Boolean.parseBoolean(data.trim());
    }

    public boolean getBoolean() {
        return data;
    }

    @Override
    public int getType() {
        return Json.BOOLEAN;
    }

    @NotNull
    @Override
    public String toString() {
        return String.valueOf(data);
    }
}

File: src/main/java/main/json/BooleanJson.kt
package main.json

/**
 *
 * @author Bunnyspa
 */
class BooleanJson(data: String) : Json {
    val boolean: Boolean = java.lang.Boolean.parseBoolean(data.trim { it <= ' ' })
    override val type: Int
        get() = Json.BOOLEAN

    override fun toString(): String {
        return boolean.toString()
    }

}